--- origsrc/telepathy-qt4-0.2.1/TelepathyQt4/Farsight/Makefile.am	2009-12-04 09:46:57.000000000 -0600
+++ src/telepathy-qt4-0.2.1/TelepathyQt4/Farsight/Makefile.am	2010-01-25 23:14:24.838499100 -0600
@@ -18,6 +18,7 @@ lib_LTLIBRARIES = libtelepathy-qt4-farsi
 # that vim quickfix mode (:make) doesn't interpret the libtool --mode=link
 # command as an error message in a bizarrely named file
 libtelepathy_qt4_farsight_la_LDFLAGS = \
+	-no-undefined \
     -version-info "$(LT_CURRENT)":"$(LT_REVISION)":"$(LT_AGE)"
 libtelepathy_qt4_farsight_la_LIBADD = \
 	$(QTCORE_LIBS) \
--- origsrc/telepathy-qt4-0.2.1/TelepathyQt4/Makefile.am	2009-12-04 09:01:45.000000000 -0600
+++ src/telepathy-qt4-0.2.1/TelepathyQt4/Makefile.am	2010-01-25 23:11:59.699197600 -0600
@@ -45,6 +45,7 @@ lib_LTLIBRARIES = libtelepathy-qt4.la
 # that vim quickfix mode (:make) doesn't interpret the libtool --mode=link
 # command as an error message in a bizarrely named file
 libtelepathy_qt4_la_LDFLAGS = \
+	-no-undefined \
     -version-info "$(LT_CURRENT)":"$(LT_REVISION)":"$(LT_AGE)"
 libtelepathy_qt4_la_LIBADD = $(ALL_LIBS)
 libtelepathy_qt4_la_DEPENDENCIES = Makefile.am
--- origsrc/telepathy-qt4-0.2.1/tests/lib/Makefile.am	2009-12-03 10:27:07.000000000 -0600
+++ src/telepathy-qt4-0.2.1/tests/lib/Makefile.am	2010-01-25 23:47:09.169852500 -0600
@@ -24,7 +24,7 @@ CLEANFILES = $(BUILT_SOURCES)
 libtp_qt4_tests_la_SOURCES = \
     test.cpp \
     test.h
-libtp_qt4_tests_la_LIBADD = $(top_builddir)/TelepathyQt4/libtelepathy-qt4.la
+libtp_qt4_tests_la_LIBADD = $(top_builddir)/TelepathyQt4/libtelepathy-qt4.la $(QTTEST_LIBS)
 
 if ENABLE_TP_GLIB_TESTS
 
