--- origsrc/telepathy-qt4-0.4.1/TelepathyQt4/CMakeLists.txt	2010-12-08 10:20:44.000000000 -0600
+++ src/telepathy-qt4-0.4.1/TelepathyQt4/CMakeLists.txt	2010-12-19 21:46:08.768127000 -0600
@@ -510,6 +510,7 @@ endforeach(moc_src ${telepathy_qt4_MOC_S
 target_link_libraries(telepathy-qt4
     ${QT_QTCORE_LIBRARY}
     ${QT_QTDBUS_LIBRARY}
+    ${QT_QTXML_LIBRARY}
     ${QT_QTNETWORK_LIBRARIES})
 
 if (ENABLE_COMPILER_COVERAGE)
--- origsrc/telepathy-qt4-0.4.1/TelepathyQt4/Farsight/CMakeLists.txt	2010-12-08 10:20:44.000000000 -0600
+++ src/telepathy-qt4-0.4.1/TelepathyQt4/Farsight/CMakeLists.txt	2010-12-19 21:46:46.460194300 -0600
@@ -29,6 +29,7 @@ if(FARSIGHT_COMPONENTS_FOUND)
         ${QT_QTDBUS_LIBRARY}
         ${QT_QTCORE_LIBRARY}
         ${TELEPATHY_FARSIGHT_LIBRARIES}
+        ${PC_TELEPATHY_GLIB_LIBRARIES}
         ${GSTREAMER_INTERFACE_LIBRARY}
         telepathy-qt4)
 
--- origsrc/telepathy-qt4-0.4.1/tests/lib/glib/CMakeLists.txt	2010-12-08 10:20:44.000000000 -0600
+++ src/telepathy-qt4-0.4.1/tests/lib/glib/CMakeLists.txt	2010-12-19 21:40:35.642108900 -0600
@@ -40,7 +40,7 @@ if(ENABLE_TP_GLIB_TESTS)
         util.c
         util.h)
     add_library(tp-glib-tests SHARED ${tp_glib_tests_SRCS})
-    target_link_libraries(tp-glib-tests ${TELEPATHY_GLIB_LIBRARIES}
+    target_link_libraries(tp-glib-tests
                           example-cm-callable
                           example-cm-contactlist
                           example-cm-csh
@@ -49,6 +49,7 @@ if(ENABLE_TP_GLIB_TESTS)
                           tp-glib-tests-future-extensions
                           future-example-cm-call
                           future-example-cm-conference
-                          future-example-cm-conference-draft)
+                          future-example-cm-conference-draft
+                          ${PC_TELEPATHY_GLIB_LIBRARIES})
 
 endif(ENABLE_TP_GLIB_TESTS)
