--- origsrc/telepathy-qt-0.9.3/tests/lib/glib/CMakeLists.txt	2012-07-13 12:40:23.000000000 -0500
+++ src/telepathy-qt-0.9.3/tests/lib/glib/CMakeLists.txt	2013-08-30 01:00:27.656228700 -0500
@@ -57,17 +57,17 @@ if(ENABLE_TP_GLIB_TESTS)
     endif(ENABLE_TP_GLIB_GIO_TESTS)
     add_library(tp-glib-tests SHARED ${tp_glib_tests_SRCS})
     target_link_libraries(tp-glib-tests
-        ${TELEPATHY_GLIB_LIBRARIES}
-        ${GLIB2_LIBRARIES}
-        ${GOBJECT_LIBRARIES}
-        ${GIO_LIBRARIES}
-        ${DBUS_GLIB_LIBRARIES}
-        ${DBUS_LIBRARIES}
         example-cm-call
         example-cm-callable
         example-cm-contactlist
         example-cm-echo
         example-cm-echo2
-        tp-glib-tests-future-extensions)
+        tp-glib-tests-future-extensions
+        ${TELEPATHY_GLIB_LIBRARIES}
+        ${GLIB2_LIBRARIES}
+        ${GOBJECT_LIBRARIES}
+        ${GIO_LIBRARIES}
+        ${DBUS_GLIB_LIBRARIES}
+        ${DBUS_LIBRARIES})
 
 endif(ENABLE_TP_GLIB_TESTS)
--- origsrc/telepathy-qt-0.9.3/tests/lib/glib-helpers/CMakeLists.txt	2012-07-13 12:40:23.000000000 -0500
+++ src/telepathy-qt-0.9.3/tests/lib/glib-helpers/CMakeLists.txt	2013-08-30 00:59:59.307607300 -0500
@@ -18,5 +18,6 @@ if(ENABLE_TP_GLIB_TESTS)
         ${DBUS_GLIB_LIBRARIES}
         ${QT_QTCORE_LIBRARY}
         ${QT_QTDBUS_LIBRARY}
+        ${QT_QTTEST_LIBRARY}
         telepathy-qt${QT_VERSION_MAJOR})
 endif(ENABLE_TP_GLIB_TESTS)
